### PR TITLE
fix: disable Zend Signals for ZTS builds

### PR DESCRIPTION
**Pull request by dunglas**
Zend Signals are useless and cause segmentation faults when using ZTS:
* https://github.com/php/php-src/issues/9649#issuecomment-1264330874
* https://github.com/php/php-src/pull/5591#issuecomment-650064098

This patch disables Zend Signals for ZTS builds.

### CODE REVIEW

File: Dockerfile-linux.template [hunk 1]
```diff
@@ -392,6 +392,8 @@ RUN set -eux; \
 {{ ) else ( -}}
 		--enable-maintainer-zts \
 {{ ) end -}}
+# https://github.com/php/php-src/issues/9649#issuecomment-1264330874
```
Comment (tianon) [inline Dockerfile-linux.template:395]: I think it would probably be useful for this to also link to https://externals.io/message/118859, where you've given a very nice summary of the situation (with a lot of additional links). :innocent:

Comment (dunglas) [reply Dockerfile-linux.template]: Link updated!

File: 8.2-rc/alpine3.16/zts/Dockerfile [hunk 2]
```diff
@@ -163,6 +163,8 @@ RUN set -eux; \
 		--disable-cgi \
 		\
 		--enable-zts \
+# https://externals.io/message/118859
```
Comment (withinboredom) [inline 8.2-rc/alpine3.16/zts/Dockerfile:166]: FYI `--enable-embed` is missing from this Dockerfile, but doesn't look related to this PR.

Comment (yosifkit) [reply 8.2-rc/alpine3.16/zts/Dockerfile:166]: It is only enabled on some variants to minimize the relative size increase (which for alpine based images is significant).

https://github.com/docker-library/php/pull/939#issuecomment-733970326: 
> In coming back to this, I think the question of whether to add this to the CLI variant boils down to the use case of the CLI variants, and whether they'll be substantially affected by a ~14MB increase in size (which in Alpine's case, is almost a 20% size increase).
>
>[...] we start by adding this to the Debian-based CLI variants (so that the Alpine variants are still as small as is reasonably possible)

https://github.com/docker-library/php/pull/1208#discussion_r726497221:
> it's a ~14MB increase. On the Debian-based images, we're on the order of almost 500MB already, so another 14 is pretty negligible in comparison. On Alpine, we're currently at ~84MB, so this would be a pretty noticeable increase in overall image size.

Comment (withinboredom) [reply 8.2-rc/alpine3.16/zts/Dockerfile:166]: Thanks for the background @yosifkit!

Comment (dunglas) [reply 8.2-rc/alpine3.16/zts/Dockerfile:166]: But are ZTS builds useful for other things than using PHP a library? I mean, this is probably one of the most popular use case.

As we provide Alpine variants in FrankenPHP, having this flag set would allow us (and probably NGINX Unit too) to use the official images for Alpine too.
